cs/src/core/Index/FASTER/FASTERImpl.cs

```diff
@@ -762,10 +845,32 @@ internal enum LatchOperation : byte
                 {
                     // ah, old record slipped onto disk
                     hlog.GetInfo(newPhysicalAddress).Invalid = true;
+                    psfArgs.ChangeTracker = null;
                     status = OperationStatus.RETRY_NOW;
                     goto LatchRelease;
                 }
 
+                if (this.PSFManager.HasPSFs)
```

Comment:
Move this to PSF.dll - make sure that these code fragments are isolated from rest of code so that future merges are easy.

Reply:
Can you make these code chunks a call to a method (in a PSF-specific directory), so that the main Internal* Logic are largely similar, making future merges from master easier? Instead of inlining all the stuff inside here.